generators/client-2/templates/src/main/webapp/app/blocks/handlers/_state.handler.ts

```diff
@@ -1,66 +1,52 @@
 import { VERSION, DEBUG_INFO_ENABLED } from '../../app.constants';
-import { Transition } from 'ui-router-ng2';
-declare var SystemJS;
+import { TransitionService, Transition } from 'ui-router-ng2';
 
-StateHandler.$inject = ['$rootScope', '$transitions', <% if (enableTranslation) { %>/*'<%=jhiPrefixCapitalized%>LanguageService',*/<% } %> '$window',
-        /*'Auth', 'Principal',*/ '$uiRouter', '$trace'];
-
-export function StateHandler($rootScope, $transitions, <% if (enableTranslation) { %>/*<%=jhiPrefixCapitalized%>LanguageService,*/<% } %> $window,
-    /*Auth, Principal,*/ $uiRouter, $trace) {
-
-    if (DEBUG_INFO_ENABLED) {
-        $trace.enable('TRANSITION');
-        SystemJS.import('ui-router-visualizer').then(vis => vis.visualizer($uiRouter));
-    }
+StateHandler.$inject = ['$rootScope'];
+export function StateHandler($rootScope) {
 
     return {
         initialize: initialize
     };
 
     function initialize() {
         $rootScope.VERSION = VERSION;
-        let deregistrationFns = [];
+    }
+}
 
-        deregistrationFns.push($transitions.onStart({}, (transition: Transition) => {
-            $rootScope.toState = transition.to();
-            $rootScope.toParams = transition.params();
-            $rootScope.fromState = transition.from();
+export function registerTransitionHooks($transitions: TransitionService) {
+    $transitions.onStart({}, (transition: Transition) => {
+        let $rootScope = transition.injector().get('$rootScope');
+        $rootScope.toState = transition.to();
+        $rootScope.toParams = transition.params();
+        $rootScope.fromState = transition.from();
 
-            /*if (Principal.isIdentityResolved()) { //TODO needs to fixed after migration
-                Auth.authorize();
-            }*/
+        /*if (Principal.isIdentityResolved()) { //TODO needs to fixed after migration
+         Auth.authorize();
```

Comment:
I can provide assistance on overhauling the authentication code.  Look at the sample app for an example.  

https://github.com/ui-router/sample-app-ng1/blob/master/app/global/auth.service.js - the authentication service

https://github.com/ui-router/sample-app-ng1/blob/master/app/global/requiresAuth.hook.js - the requires authentication hook

https://github.com/ui-router/sample-app-ng1/blob/master/app/main/app.states.js#L44-L83 - the login state and resolve, which determines where to return to after a successful authentication